### ISSUE TITLE
why channel models is not finetuned with other tuning approaches ? 

### ISSUE THREAD

**Issue by dorost1234** [shmsw25/Channel-LM-Prompting#1]
Hi 
Great work! could you elaborate why only direct model is tuned with head-tune and transform-tune options but not the channel variant? 

thanks!

**Comment by shmsw25** [comment]
Hi, great question. It was because of two reasons: (1) we mainly care about methods with comparable number of trainable parameters. Prompt tuning (both direct and channel) and direct head tuning have similar number of trainable parameters `hidden_dimension * C`, where `C` is usually between 2 to 20. Channel head tuning has `hidden_dimension * vocab_size`, where `vocab_size` is usually a few tens of thousands. So they are not comparable. (2) In fact, even though channel head tuning has more trainable parameters, their performance is always worse than all other methods. This is probably because head tuning is competitive due to sensitivity to the verbalizers, which is only the case for direct models.

In short, we were mainly interested in prompt tuning because it is a way to use the LM as it is, but we added head tuning and transformation tuning only as even stronger direct baselines.

Let me know if you have further questions.

**Comment by dorost1234** [comment]
Thanks for the insightful comment.
